vignettes/bisg.Rmd

```diff
@@ -3,53 +3,80 @@ title: "Bayesian Improved Surname Geocoding (BISG)"
+  
+In R, the package that performs BISG is called, WRU: Who Are You `wru` [https://cran.r-project.org/web/packages/wru/index.html]. This vignette will walk you through how to prepare your geocoded voter file for performing BISG by stepping you throught the processing of cleaning your voter file, prepping voter data for running the BISG, and finally, performing BISG to obtain racial/ethnic probailities of individuals in a voter file.
 
 ## *Performing BISG on your data*
-The first step in performing BISG is to geocode your voter file addresses. For information on geocoding, visit the Geocoding Vignette. 
+We will perform BISG using the previous Gwinnett and Fulton county voter registration data called `ga_geo.csv` that was geocoded in the **eiCompare: Geocoding vignette**. 
 
-In this tutorial, we will be using the East Ramapo, New York voter file called, **voter_file_geocoded**, that consists of indivduals registering to vote during the year of 2016. 
+The first step in performing BISG is to geocode your voter file addresses. For information on geocoding, visit the Geocoding Vignette. 
 
 Let's begin by loading your geocoded voter data into R/RStudio.
 
 ### Step 1: Load R libraries/packages, voter file, and census data
-
 Load the R packages needed to perform BISG. If you have not already downloaded the following packages, please install these packages.
+
 ```{r}
-# Load libraries/packages
-
-suppressMessages(c(
-  library(devtools),
-  install.packages("tidyverse"),
-  library(tidyverse),
-  install.packages("stringr"),
-  library(stringr),
-  install.packages("sf"),
-  library(sf),
-  install.packages("eiCompare"),
-  library(eiCompare),
-  library(wru),
+# Install eiCompare
+suppressPackageStartupMessages({
+  devtools::install_github("RPVote/eiCompare@master")
+})
+```
+
+```{r}
+# Load libraries
+suppressPackageStartupMessages({
+  library(devtools)
+  library(tidyverse)
+  library(stringr)
+  library(tigris)
+  library(leaflet)
+  library(sf)
+  library(eiCompare)
+  library(wru)
   library(readr)
-))
+})
+```
+
+```{r}
+# source files
+source("~/github/eiCompare/R/wru_predict_race_wrapper.R")
+source("~/github/eiCompare/R/voter_file_utils.R")
+```
+
+Load in census data, the shape_file and geocoded voter registation data with latitude and longitude coordinates Gwinnett and Fulton .
+
+Make sure to load your census data that details certain geographies (i.e. counties, cities, tracts, blocks, etc.)
+```{r}
+# Load Georgia census data
+path_census <- "~/shared/georgia/"
+census_data <- readRDS(paste(path_census, "georgia_census.rds", sep = ""))
+```
+
+Next, load the state shape file using the sf::st_read() function.
+```{r, echo=FALSE}
+# Load Georgia block shape file
+shape_file <- blocks(state = "GA", county = c("Gwinnett", "Fulton"))
```

Comment:
i am assuming this is using `tigris`? if so can you add the namespace so it's clear?

is CMD check not running the vignette? not sure how a user will be able to run this code without having a Census API key in their environment

Reply:
Yes, we are using the tigris package and actually the Census Bureau's Geocoding API is publicly available so no API key is needed. https://www.census.gov/data/developers/guidance/api-user-guide.html  

However, I can explicitly explain this before loading the shape file using tigris.